src/Updater.php

```diff
@@ -320,13 +321,28 @@ protected function hasPubKey()
         return $this->hasPubKey;
     }
 
+    /**
+     * Compares local and remote versions. If version strings do not follow
+     * Semver, i.e. hashes, the strings are just checked for equality.
+     * @return bool
+     */
     protected function newVersionAvailable()
     {
         $this->newVersion = $this->strategy->getCurrentRemoteVersion($this);
         $this->oldVersion = $this->strategy->getCurrentLocalVersion($this);
 
-        if (!empty($this->newVersion) && ($this->newVersion !== $this->oldVersion)) {
-            return true;
+        try {
+            if (!empty($this->newVersion)
+                && !VersionParser::equals($this->newVersion, $this->oldVersion)
+            ) {
+                return true;
+            }
+        } catch (\UnexpectedValueException $e) {
```

Comment:
UnexpectedValueException thrown by Composer\Semver. I should add a check that we do the following `!==` comparison only if the strategy in play is NOT github. Otherwise, we'll silently fail on invalid semver when there MUST be a valid semver version string.